src/process_utility.c

```diff
@@ -1688,6 +1687,33 @@ process_altertable_end_subcmd(Hypertable *ht, Node *parsetree, ObjectAddress *ob
 		case AT_DropCluster:
 			foreach_chunk(ht, process_altertable_chunk, cmd);
 			break;
+		case AT_SetTableSpace:
+			{
+				Oid			tspc_oid = get_rel_tablespace(ht->main_table_relid);
+				NameData	tspc_name;
+				Tablespaces *tspcs;
+
+				namestrcpy(&tspc_name, get_tablespace_name(tspc_oid));
+				Assert(OidIsValid(tspc_oid));
+
+				/*
+				 * this is called after postgres ALTER TABLE so command
+				 * tablespace name
+				 */
+				/* is same as current relation tablespace name */
+				Assert(strcmp(cmd->name, NameStr(tspc_name)) == 0);
```

Comment:
should just use cmd->name then in namestrcpy above. No need to do the lookup at get_tablespace_name(tspc_oid).